program-runtime/src/loaded_programs.rs

```diff
@@ -382,6 +382,19 @@ impl LoadedProgram {
     }
 
     pub fn to_unloaded(&self) -> Option<Self> {
+        #[cfg(test)]
+        if !matches!(&self.program, LoadedProgramType::TestLoaded(_)) {
+            return None;
+        }
+        #[cfg(not(test))]
+        if !matches!(
```

Comment:
ban the use of `matches!`. Spell out all the variants, so you're forced to spell
out the ones you don't want to unload, and it's more future proof and the
compiler will yell at us if we modify the variants and don't handle new cases.

Reply:
What you want to say is not to use the `_` default case. `matches!` and `match` are the same otherwise.